cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -0,0 +1,166 @@
+from cg.apps.demultiplex.sample_sheet.sample_models import FlowCellSample
+from cg.apps.demultiplex.sample_sheet.sample_sheet_validator import SampleSheetValidator
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.apps.lims import LimsAPI
+from cg.apps.lims.sample_sheet import get_flow_cell_samples
+from cg.constants.constants import FileFormat
+from cg.constants.demultiplexing import BclConverter
+from cg.exc import FlowCellError, HousekeeperFileMissingError, SampleSheetError
+from cg.io.controller import WriteFile, WriteStream
+from cg.meta.demultiplex.housekeeper_storage_functions import (
+    add_and_include_sample_sheet_path_to_housekeeper,
+)
+from cg.models.flow_cell.flow_cell import FlowCellDirectoryData
+
+LOG = logging.getLogger(__name__)
+
+
+class SampleSheetAPI:
+    """Sample Sheet API class."""
+
+    def __init__(self, flow_cell_dir: str, hk_api: HousekeeperAPI, lims_api: LimsAPI) -> None:
+        self.flow_cell_runs_dir = Path(flow_cell_dir)
+        self.hk_api: HousekeeperAPI = hk_api
+        self.lims_api: LimsAPI = lims_api
+        self.dry_run: bool = False
+        self.force: bool = False
+        self.validator = SampleSheetValidator()
+
+    def set_dry_run(self, dry_run: bool) -> None:
+        """Set dry run."""
+        LOG.debug(f"Set dry run to {dry_run}")
+        self.dry_run = dry_run
+
+    def set_force(self, force: bool) -> None:
+        """Set force."""
+        LOG.debug(f"Set force to {force}")
+        self.force = force
+
+    def get_flow_cell(self, flow_cell_name: str, bcl_converter: str) -> FlowCellDirectoryData:
+        """
+        Return a flow cell given a path and the bcl converter.
+        Raises:
+            SampleSheetError: If the flow cell directory or the data it contains is not valid.
+        """
+        flow_cell_path: Path = Path(self.flow_cell_runs_dir, flow_cell_name)
+        if not flow_cell_path.exists():
+            message: str = f"Could not find flow cell {flow_cell_path}"
+            LOG.warning(message)
+            raise SampleSheetError(message)
+        try:
+            flow_cell = FlowCellDirectoryData(
+                flow_cell_path=flow_cell_path, bcl_converter=bcl_converter
+            )
+        except FlowCellError as error:
+            raise SampleSheetError from error
+        return flow_cell
+
+    def get_valid_sample_sheet_path(self, sample_sheet_path: Path) -> Path | None:
+        """Return the sample sheet path if it exists and if it passes validation."""
+        if sample_sheet_path and sample_sheet_path.exists():
+            try:
+                self.validator.validate_sample_sheet_from_file(sample_sheet_path)
+            except (SampleSheetError, ValidationError):
+                LOG.warning(f"Sample sheet {sample_sheet_path} was not valid")
+                return
+            return sample_sheet_path
+        else:
+            LOG.warning(f"Sample sheet with path {sample_sheet_path} does not exist")
```

Comment:
should this only be warnings? If the validation fails an error would make sense to me.

Reply:
If the sample sheet does not exist in any location it simply creates it. It is rather to inform the user. This is actually the current functionality